### PR TITLE
Late Join Ghost Role Things

### PR DESCRIPTION
**Pull request by Geevies**
* Added available unique ghost spawner roles to the late join menu.
* Added station alert level to the late join menu.

image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by alsoandanswer** [comment]
Since you're here, maybe you can put the name of the character you're playing on the Customize Character/Ready/Join up/Spectate menu? At the top

**Comment by Geevies** [comment]
image
Note: the above says "Welcome, [character_name]." now, not Selected Character:

image
Changed it to not list all the roles, only say that unique ones are available.

**Comment by NonQueueingMatt** [comment]
I don't really like the in-your-face style of the join game ghost role report. What I'd do is just put the ghost spawner button in the new player options and let people see the current ghost spawners.

Also not exactly a fan of this telling you about ERTs and such.

**Comment by skull132** [comment]
> 
> 
> I don't really like the in-your-face style of the join game ghost role report. What I'd do is just put the ghost spawner button in the new player options and let people see the current ghost spawners.
> 
> Also not exactly a fan of this telling you about ERTs and such.

I'm sorta the opposite. I think people should be encouraged to pick those roles, as they're arguably more necessary for the round. Well, _some_ anyways.

**Comment by Geevies** [comment]
well whatever you want, boys, i can do it.
atm it's like, the "GHOST ROLE THINGY" shows all the time, BUT the bold green text only shows if a UNIQUE ghost role is available. so the green text won't show when you can only join as a rat or a drone. meaning, if there's a role available, the players will be prompted to investigate by the green text and then go look at the roles. yes.

File: code/modules/ghostroles/spawner/simplemob/maintdrone.dm [hunk 1]
```diff
@@ -2,6 +2,7 @@
 	short_name = "maintdrone"
 	name = "Maintenence Drone"
 	desc = "Maintain and Improve the Systems on the Aurora"
+	show_on_job_select = FALSE
```
Comment (NonQueueingMatt) [inline code/modules/ghostroles/spawner/simplemob/maintdrone.dm]: Do this for ERT too.

Comment (Geevies) [reply code/modules/ghostroles/spawner/simplemob/maintdrone.dm]: nah, i don't think removing them is a good idea, as discussed on discord

File: code/controllers/subsystems/ghostroles.dm [hunk 2]
```diff
@@ -137,6 +137,9 @@
 		if(cant_spawn)
 			to_chat(usr, "Unable to spawn: [cant_spawn]")
 			return
+		if(istype(usr, /mob/abstract/new_player))
```
Comment (skull132) [inline code/controllers/subsystems/ghostroles.dm]:
```suggestion
		if(isnewplayer(usr))
```

File: code/modules/ghostroles/spawner/base.dm [hunk 3]
```diff
@@ -66,7 +67,7 @@
 	var/cant_see = cant_see(user)
 	if(cant_see) //If we cant see it, we cant spawn it
 		return cant_see
-	if(!istype(user, /mob/abstract/observer))
+	if(!(istype(user, /mob/abstract/observer) || istype(user, /mob/abstract/new_player)))
```
Comment (skull132) [inline code/modules/ghostroles/spawner/base.dm]:
```suggestion
	if(!(istype(user, /mob/abstract/observer) || isnewplayer(user)))
```

**Pull request merged**